test/demo/contextChannels/join.inttest.ts

```diff
@@ -0,0 +1,446 @@
+            const defaultChannel = await fdc3Remote.getChannelById(testManagerIdentity, 'default');
+
+            await expect(defaultChannel.getMembers()).resolves.toContainEqual({uuid: testAppInDirectory1.uuid, name: testAppInDirectory1.name});
+        });
+
+        test('After closing the FDC3 app, result does not contains the FDC3 app', async () => {
+            const defaultChannel = await fdc3Remote.getChannelById(testManagerIdentity, 'default');
+
+            await fin.Application.wrapSync(testAppInDirectory1).quit(true);
+
+            await expect(defaultChannel.getMembers()).resolves.toEqual([testManagerIdentity]);
+        });
+    });
+
+    describe('When a non-directory app has been started', () => {
+        beforeEach(async () => {
+            await fin.Application.startFromManifest(testAppNotInDirectory.manifestUrl);
+        }, appStartupTime);
+
+        afterEach(async () => {
+            await fin.Application.wrapSync(testAppNotInDirectory).quit(true);
+        });
+
+        test('When the channel is the default channel, result contains the non-directory app', async () => {
+            const defaultChannel = await fdc3Remote.getChannelById(testManagerIdentity, 'default');
+
+            await expect(defaultChannel.getMembers()).resolves.toContainEqual({uuid: testAppNotInDirectory.uuid, name: testAppNotInDirectory.name});
+        });
+    });
+
+    describe('When a non-FDC3 app has been started', () => {
+        beforeEach(async () => {
+            await fin.Application.startFromManifest(testAppNotFdc3.manifestUrl);
+        }, appStartupTime);
+
+        afterEach(async () => {
+            await fin.Application.wrapSync(testAppNotFdc3).quit(true);
+        });
+
+        test('Result does not contain the non-FDC3 app', async () => {
+            const defaultChannel = await fdc3Remote.getChannelById(testManagerIdentity, 'default');
+
+            await expect(defaultChannel.getMembers()).resolves.toEqual([testManagerIdentity]);
+        });
+    });
+});
+
+describe('When listening for a channel-changed event', () => {
+    const listeningApp = testAppInDirectory1;
+
+    beforeEach(async () => {
+        await fdc3Remote.open(testManagerIdentity, listeningApp.name);
+    }, appStartupTime);
+
+    afterEach(async () => {
+        await fin.Application.wrapSync(listeningApp).quit(true);
+
+        const inDirectoryApp = fin.Application.wrapSync(testAppInDirectory2);
+        if (await inDirectoryApp.isRunning()) {
+            await inDirectoryApp.quit(true);
+        }
+
+        const notInDirectoryApp = fin.Application.wrapSync(testAppNotInDirectory);
+        if (await notInDirectoryApp.isRunning()) {
+            await notInDirectoryApp.quit(true);
+        }
+
+        const notFdc3App = fin.Application.wrapSync(testAppNotFdc3);
+        if (await notFdc3App.isRunning()) {
+            await notFdc3App.quit(true);
+        }
+    });
+
+    type TestParam = string | Identity | (() => Promise<any>);
```

Comment:
Can this be defined as a tuple?.. To keep the type-safety without unions/casts? Kinda depends how `each` is defined, I guess.

```
type TestParam = [string, Identity, () => Promise<any>];
```

Reply:
Huh, I'm annoyed because I was sure I tried this. Will fix!